src/solver/fft.cpp

```diff
@@ -102,34 +107,91 @@ static void cgemm_grid(size_t* global_work_size,
     global_work_size[1] = totalWorkGroups1 * local_work_size[1];
 }
 
-static int FindFFTKernel(const Handle& handle,
-                         const TensorDescriptor& xDesc,
-                         const TensorDescriptor& wDesc,
-                         const TensorDescriptor& yDesc,
-                         size_t workSpaceSize,
-                         std::vector<KernelInvoke>& kernels,
-                         bool fwd,
-                         const NetworkConfig& kcache_key)
+bool FFT::IsApplicable(const ConvolutionContext& ctx) const
 {
-
-    if(workSpaceSize == 0)
-        return -1;
-
     // disable running any FFT based convolutions by checking this env variable
     if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_FFT{}))
-        return -1;
+        return false;
+
+    const auto is_fwd    = ctx.direction.IsForward();
+    decltype(auto) conv  = ctx.conv_problem.GetConv();
+    decltype(auto) xDesc = is_fwd ? ctx.conv_problem.GetIn() : ctx.conv_problem.GetOut();
+    decltype(auto) yDesc = is_fwd ? ctx.conv_problem.GetOut() : ctx.conv_problem.GetIn();
+    decltype(auto) wDesc = ctx.conv_problem.GetWeights();
 
     if(xDesc.GetType() != miopenFloat || wDesc.GetType() != miopenFloat ||
        yDesc.GetType() != miopenFloat)
-        return -1;
+        return false;
```

Comment:
```suggestion
if(!ctx.conv_problem.IsFp32())
    return false;
```

Reply:
I'll fix that. By the way, you can make a multi-line suggestion by dragging the + image.